packages/react/src/date-picker/utils.ts

```diff
@@ -90,9 +91,17 @@ export function transformValuePropToInputValue(
 	valueProp: Date | string | undefined,
 	dateFormat: AcceptedDateFormats
 ): string {
-	if (typeof valueProp === 'string') return valueProp;
+	if (
```

Comment:
Is this the same but cleaner?

I used `normaliseDateString` earlier and removed the `typeof undefined` checks because they felt unnecessary?

```
export function transformValuePropToInputValue(
	valueProp: Date | string | undefined,
	dateFormat: AcceptedDateFormats
): string {
	if (valueProp === undefined) return '';

	const valueAsDate =
		typeof valueProp === 'string' ? normaliseDateString(valueProp) : valueProp;

	if (valueAsDate === undefined) return '';

	if (isValidDate(valueAsDate)) return formatDate(valueAsDate, dateFormat);

	return '';
}
```

Reply:
Invalid ISO strings are converted to `''` with this update, rather than being left as-entered. We may have to leave it as-is.

Reply:
Ah, so it should be as simple as changing `if (valueAsDate === undefined) return '';` to `if (valueAsDate === undefined) return valueProp;` then?

Reply:
Sure would be nice to have tests for this functiom

Reply:
There are tests https://github.com/agriculturegovau/agds-next/blob/develop/packages/react/src/date-picker/utils.test.ts#L156

Reply:
With this PR, we want to split how we handle strings, i.e. we keep the old behaviour of returning the string as-is, i.e. this check `typeof valueProp === 'string'` previously just returned the string as it was entered. Now though, if we can parse it as an ISO date string, we convert it to a date object that will play nicely with the component.

Reply:
Yay, tests! So my latest suggestion passes all of them, but TypeScript is being TypeScript. If I'm testing it's `undefined`, how could it be a string?! 😡  
<img width="726" alt="image" src="https://github.com/user-attachments/assets/fb27434c-1f9f-4b4c-8512-321fdf1777b9">

Reply:
Oh it's `valueProp` that could be a date… hmmmm…

Reply:
What do you think? Any further thoughts?

Reply:
How about this which appeases TS and passes all tests? It feels cleaner and clearer to me, and only parses once

```
export function transformValuePropToInputValue(
	valueProp: Date | string | undefined,
	dateFormat: AcceptedDateFormats
): string {
	if (valueProp === undefined) return '';

	const valueAsDateOrUndefined =
		typeof valueProp === 'string' ? normaliseDateString(valueProp) : valueProp;

	if (valueAsDateOrUndefined === undefined) return valueProp.toString();

	if (isValidDate(valueAsDateOrUndefined))
		return formatDate(valueAsDateOrUndefined, dateFormat);

	return '';
}
```

Reply:
Ok, sounds good. I'll update.